### ISSUE TITLE
Swift naming conflict for +acknowledgementsViewController

### ISSUE THREAD

**Issue by chrisballinger** [vtourraine/VTAcknowledgementsViewController#41]
It seems that by Swift is clobbering the init method and makes it impossible to call the required convenience initializer.

```swift
let ackVC = VTAcknowledgementsViewController.acknowledgementsViewController()
```

Results in this error: 'acknowledgementsViewController()' is unavailable: use object construction 'VTAcknowledgementsViewController()'

If you do what the compiler suggests, Swift calls `-init` instead of `+acknowledgementsViewController`, and you end up with empty acknowledgements.

**Comment by chrisballinger** [comment]
There is also not a great workaround because all of the built-in ways of grabbing the default plist are private.

**Comment by vtourraine** [comment]
Now that’s problematic, thanks for the heads-up.

There should be some sort of convention to make it work both with ObjC and Swift. If we really need to expose the methods to automatically read the default .plist, of course that’s always an option, but I’d like to avoid that if possible.

If someone can figure this one out, please let us know. I’ll give it a try later, I guess every opportunity to practice Swift is worth seizing.

**Comment by chrisballinger** [comment]
Just started toying around with it recently after waiting for the early adopters to deal with the growing pains. Not too bad!

**Comment by vtourraine** [comment]
For reference, [Swift documentation about initialization].(https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/Initialization.html)

My impression is that the current `acknowledgementsViewController` initializer should just be renamed into `init`, and the existing `initWithAcknowledgementsPlistPath:` should be marked as `NS_DESIGNATED_INITIALIZER `.

But in this case, maybe we should push that logic further, and add a new `initWithAcknowledgements:(NSArray <VTAck *> *)` method as the only designated initializer, and just have `init` and `initWithPlistPath:` as convenience initializer.

**Comment by vtourraine** [comment]
I haven’t found a good solution yet, but now there’s an alternative: AcknowList is the same library as VTAcknowledgementsViewController, re-written in Swift. It’s also available on CocoaPods (`pod 'AcknowList'`).

https://github.com/vtourraine/AcknowList

I think we still need to find a better way to bridge this Objective-C library to Swift, but for Swift projects, that should be the way forward.

**Comment by vtourraine** [comment]
Addressed with #44

Everyone, feel free to open a new issue if you have more feedback on this issue, or on Swift interoperability en general.
